### PR TITLE
`Development`: Update client dependencies to Angular 13

### PR DESCRIPTION
**Pull request by coaxial1**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).

### Motivation and Context

Angular 13 has been released and we should update.

### Description

This PR bumps Angular to v13, alongside a bunch of other dependencies. I decided to start freshly as the original PR of @krusche ( #4245 ) made a lot of problems on my machine (although I think I could have fixed it by what I learned during the creation of this PR, but well, now we're here).
Additionally, about 131 client tests failed after the update, which this PR does fix as well.

### Steps for Testing
This PR needs extensive testing in a testing session of literally all UI features. Although all tests pass, some of them had to be changed. Especially, we should review all component usages of which the tests have been changed, and all date picker and code editor usages need to be manually verified as they made a lot of problems during the upgrade as they are **extremely old and outdated** and are not even compatible with Angular 9. _We should figure out if we can replace these two dependencies in the future as they are abandoned and unmaintained since years._

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

File: src/main/webapp/app/overview/course-lectures/video-unit/video-unit.component.ts
Comment (krusche) [inline src/main/webapp/app/overview/course-lectures/video-unit/video-unit.component.ts:31]: we should still use safeResourceUrlPipe here

Comment (coaxial1) [reply src/main/webapp/app/overview/course-lectures/video-unit/video-unit.component.ts:31]: We do, I moved it to the template because it complained about it

**Comment by coaxial1** [comment]
This is currently blocked due to
https://github.com/ng-bootstrap/ng-bootstrap/issues/4178
more specifically this fix https://github.com/ng-bootstrap/ng-bootstrap/pull/4185
as we can't run tests without that being fixed
v12 is not an option as it bumps bootstrap to v5 (breaking; additionally blocked by ngx-treeview depending on v4)

Modules with incompatible / outdated deps:

_Not yet ready for angular 13, but will (hopefully) be soon:_
ngx-charts
ngx-datatable
@danielmoncada/angular-datetime-picker
ng2-charts

_Abandoned / archived deps, will probably not be updated):_
awesome-typescript-loader
cache-loader (deprecated, advises update to webpack 5)
ng-starrating
ng2-ace-editor (we might just maintain this ourselves)
ng2-dnd
ng2-material-dropdown
ngx-treeview
angular-fittext

**Comment by krusche** [comment]
I removed `cache-loader` recently, because it was not used any more.
I integrated `ng2-ace-editor` in #4441 

`ng2-material-dropdown` is a dependency of `ngx-chips` which was recently updated to 3.0.0. There are no release notes, but the commits indicate support for Angular 13

**Comment by coaxial1** [comment]
danielmoncada/angular-datetime-picker has been updated to v13 and I added it to the PR.

I think we don't need to wait for the other libraries since it seems they aren't really moving.
I'll schedule this PR for testing in the testing session on Thursday

**Comment by krusche** [comment]
I suggest we integrate the code editor library changes of my other PR first before testing this one

**Comment by coaxial1** [comment]
I'll merge it into this PR before we test it in case it's not yet merged, so we can test both at once (or rather, the combination of both) and see if anything breaks

**Review by Malyuk-A — APPROVED** [review]
Tested during testing session and locally. I could not find issues.

**Review by willergeorg — APPROVED** [review]
Tested during testing session. Functionality was fine.

**Review by rs-maker — APPROVED** [review]
Have run the cypress tests on the PR and they passed. Also tested manually and couldnt find anything wrong. Didnt look at code for obvious reasons.

**Review by akesfeden — APPROVED** [review]
Tested during testing session on TS3. Functionality seemed correct, code changes look good too.

**Review by filip-gregurevic — DISMISSED** [review]
Tested during testing session

**Review by Malyuk-A — APPROVED** [review]
re-approve 👍

**Review by willergeorg — APPROVED** [review]
Re-approve.

**Review by filip-gregurevic — APPROVED** [review]
re-approve

**Review by krusche — APPROVED** [review]

**Pull request merged**
